caffe2/core/tensor.h

```diff
@@ -952,16 +952,11 @@ class CAFFE2_API UndefinedTensorImpl final : public TensorImpl {
  //  error: identifier "at::UndefinedTensor::_singleton" is undefined in device code
  // (ostensibly because the constexpr tricks MSVC into trying to compile this
  // function for device as well).
-#ifdef _WIN32
```

Comment:
After looking at the comments [here](https://stackoverflow.com/questions/2479784/exporting-static-data-in-a-dll#comment2471695_2479944), I think we may have to move the implementation of the function `singleton` into the source file `tensor.cc`.